### PR TITLE
State RFC Tests Only - Used for Review

### PR DESCRIPTION
**Pull request by teor2345**
@yaahc I'm using this PR to just see the extra test code.

### CODE REVIEW

File: zebra-chain/src/transaction/arbitrary.rs [hunk 1]
```diff
@@ -189,16 +192,31 @@ impl Arbitrary for ShieldedData {
...
+
+        let network_upgrade = NetworkUpgrade::current(network, height);
+
+        match network_upgrade {
+            NetworkUpgrade::Genesis | NetworkUpgrade::BeforeOverwinter => Self::v1_strategy(height),
+            NetworkUpgrade::Overwinter => Self::v2_strategy(height),
+            NetworkUpgrade::Sapling => Self::v3_strategy(height),
+            NetworkUpgrade::Blossom | NetworkUpgrade::Heartwood | NetworkUpgrade::Canopy => {
+                Self::v4_strategy(height)
+            }
```
Comment (teor2345) [inline zebra-chain/src/transaction/arbitrary.rs]: This is *sweet*

File: zebra-state/src/memory_state.rs [hunk 2]
```diff
@@ -456,3 +457,135 @@ impl Ord for Chain {
...
+
+        Ok(())
+    }
+
+    #[test]
+    fn construct_many() -> Result<()> {
+        zebra_test::init();
+
+        let mut block: Arc<Block> =
+            zebra_test::vectors::BLOCK_MAINNET_434873_BYTES.zcash_deserialize_into()?;
```
Comment (teor2345) [inline zebra-state/src/memory_state.rs:534]: Do we need to use genesis here?
Or does the code we're testing not care about the chain root invariants?

```suggestion
    #[test]
    fn construct_single() -> Result<()> {
        zebra_test::init();
        let block = zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES.zcash_deserialize_into()?;

        let mut chain = Chain::default();
        chain.push(block);

        assert_eq!(1, chain.blocks.len());

        Ok(())
    }

    #[test]
    fn construct_many() -> Result<()> {
        zebra_test::init();

        let mut block: Arc<Block> =
            zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES.zcash_deserialize_into()?;
```

**Review by teor2345 — COMMENTED** [review]
Looks good so far, seems like a good design.

File: zebra-state/src/memory_state.rs (hunk 2 above)
Comment (yaahc) [reply zebra-state/src/memory_state.rs:534]: I don't think we should worry about the genesis block, this is only supposed to ever handle non-finalized data, so the genesis block should never pass through this data structure, or even any pre sapling blocks.

**Comment by teor2345** [comment]
Ok, I'm done here, more comments in #1093.

**Pull request closed without merging**